### DISCUSSION TITLE
Adding a variable after removing an import with the same name

### DISCUSSION THREAD

**Discussion by Marabyte** [swc-project/swc#6578] [Q&A] [answered]
Hi all!
I'm trying to build a plugin to something along these lines:
Given a file that is importing a css file like this
```javascript
import styles from './b.module.css';
export const b = () => (<span className={styles.foo} />);
```
I want to replace the import with a `const` like this
```javascript
const styles = {};
export const b = () => (<span className={styles.foo} />);
```

Right now I'm able to remove the CSS Modules import and then add the const with css mapping but I end up with this:
```javascript
export const b = () => (<span className={styles.foo} />);
const styles1 = {};
```

I've followed the instructions in the [Plugin cheatsheet](https://swc.rs/docs/plugin/ecmascript/cheatsheet#mark--delete) and then I'm using [`visit_mut_stmt_like`](https://github.com/sardinedev/swc-plugin-lightning-css/blob/main/src/lib.rs#L147) to add a new variable declaration with the same name as the import.
I'm guessing the `const` name is renamed because when it's added to the tree the `import` is still there.

Is there any way of controlling the order of these events?
Or, most likely, am I doing something silly here?

**Comment by kdy1** [comment, marked as answer]
Read https://rustdoc.swc.rs/swc_ecma_transforms_base/fn.resolver.html
